Controller/HeadlessWebsiteController.php

```diff
@@ -75,14 +76,64 @@ private function renderTemplateResponse(
             throw new HttpException(406, sprintf('Page does not exist in "%s" format.', $requestFormat));
         }
 
-        if ($preview) {
-            $parameters['previewParentTemplate'] = $viewTemplate;
-            $parameters['previewContentReplacer'] = Preview::CONTENT_REPLACER;
+        // if partial render only content block else full page
```

Comment:
`render only content block of twig template if clients requests partial rendering`